test/connectivity/README.md

```diff
@@ -69,3 +72,106 @@ Network offline/online should not affect this test result since Edge hub stores
+	
+##### Test Mechanism: 
+	1. The Message Sender invokes the IoT Hub to send  a message to a targeted device
+	2. The Message Sender notify the Test Result Coordinator that a message has been sent.
+	3. The IoT Hub sends a message to Edge Hub addressing a targeted device
+	4. The Edge Hub forward the message to a Message Receiver device module.
+	5. The Message Receiver notifies that the message successfully received to the Test Result Coordinator
+	6. The Message Receiver replies that the message successfully received to the Edge Hub
+	7. The Edge Hub forward the Message Receiver's response back to IoT Hub
+	8. The IoT Hub forward the response to Message Sender.
+
+##### Passing Criteria: 
+All the messages sent by Message Sender is received in order at Message Receiver within (EdgeHub_Recover_Time) + 15 minutes.
+
+#### EdgeHub Restart Test
+The goal of EdgeHub Restart Test is to verify/benchmark a restart period of an EdgeHub. The measurement is done using two methods: a Direct Method and a Message.
+
+The message test is developed to strictly test only the only when the EdgeHub complete its restart and is ready for a module to connect.  While the direct method test also make sure that the receiver can successfully response to the direct method request from the sender. This results in a slightly different time measurement between the two test (see the diagram below).
+
+![Connectivity Test Diagram](./images/ConnectivityTest_EdgeHubRestart_Timeline.png)
+
+##### EdgeHub Restart Test : Message Test
+
+![Connectivity Test Diagram](./images/ConnectivityTest_EdgeHubRestart_Message.png)
+
+###### Test Mechanism: 
+	1. The EdgeHub Restart Tester issue a direct method to EdgeAgent notifying the EdgeAgent to restart EdgeHub module
+	2. Once the restart direct method response with a success, the EdgeHub Restart Tester attempts to send a message to Message Receiver via EdgeHub. 
+		a. The SDK will throw an exception within EdgeHub Restart Tester if the EdgeHub is unabled to be connected
+		b. If the SDK stops throwing an exception and respond with HTTP OK, it is an indicator that the sender module is able to communicate with EdgeHub. However, this does not imply that the EdgeHub can successfully send the message to the Message Receiver.
+	3. The EdgeHub Restart Tester reports its { restart time, message sending response, and message sending completion time } to the Test Result Coordinator
+	4. Eventually the EdgeHub will forward the message to the Message Receiver once the receiver module can successfully reconnected to the EdgeHub
+	5. Once the receiver module receives a message, it will report the message receival to Test Result Coordinator
+
+###### Passing Criteria:
```

Comment:
should we add passing criteria for every test?

Reply:
I think it makes sense to have a passing criteria for test cases, otherwise, the test case doesn't mean anything.

Reply:
Please add passing criteria for Cloud to EdgeAgent Direct Method Test (PING) and Cloud to Module Direct Method Test.  thanks.